src/lib/ggshield-resolver.ts

```diff
@@ -58,24 +58,12 @@ export class GGShieldResolver {
   async testConfiguration(
     configuration: GGShieldConfiguration
   ): Promise<void> {
-    let proc = runGGShieldCommand(configuration, ["quota"]);
-    if (proc.error || proc.stderr.length > 0) {
-      if (proc.error) { 
-        if (proc.error.message.includes("ENOENT")) {
-          throw new Error(
-            `GGShield path provided in settings is invalid: ${configuration.ggshieldPath}.`
-          );
-        } else {
-          throw new Error(proc.error.message);
-        }
-      } else if (proc.stderr.includes("Invalid API key")) {
-        throw new Error(
-          `API key provided in settings is invalid.`
-        );
-      }
-    } else {
-      this.configuration = configuration;
-      return;
+    // Check if the ggshield path is valid
+    let proc = runGGShieldCommand(configuration, ["--version"]);
+    if (proc.status !== 0) {
+      window.showErrorMessage(`GitGuardian: Invalid ggshield path. ${proc.stderr}`);
+      throw new Error(proc.stderr);
```

Comment:
Is knowing if the API key valid not necessary because we will get the same error when trying to scan ?